packages/ifpack/src/Ifpack_Hypre.cpp

```diff
@@ -624,4 +616,26 @@ int Ifpack_Hypre::CreatePrecond(){
   return (this->*PrecondCreatePtr_)(comm, &Preconditioner_);
 } //CreatePrecond()
 
+
+//==============================================================================
+int Ifpack_Hypre::CopyEpetraToHypre(){
+  for(int i = 0; i < A_->NumMyRows(); i++){
+    int numElements;
+    IFPACK_CHK_ERR(A_->NumMyRowEntries(i,numElements));
+    std::vector<int> indices; indices.resize(numElements);
+    std::vector<double> values; values.resize(numElements);
+    int numEntries;
+    IFPACK_CHK_ERR(A_->ExtractMyRowCopy(i, numElements, numEntries, &values[0], &indices[0]));
```

Comment:
Ifpack depends on Teuchos and Teuchos requires C++ >= 11, so you can and should use `values.data()` and `indices.data()` here.  Some C++ Standard Library implementations have a bounds checking option that throws an exception in `std::vector::operator[]` if out of bounds, even if one doesn't actually use that entry.

Reply:
Done in next push.